ExcelWriter.py

```diff
@@ -1,130 +1,228 @@
     def __init__(self, name):
             self.workbook = xlsxwriter.Workbook(name)
             self.worksheet = self.workbook.add_worksheet()
-            self.currentRow = 0
-            self.endRow = 0
-            self.currentCol = 0            
-            self.default_format = self.workbook.add_format()
-            self.default_format.set_font_size(11)
-            self.default_format.set_text_wrap()
-            #self.default_format.set_shrink()
-            self.default_format.set_align('top')
-            self.maxCol = []
-            self.indentFormat = {}
+            self.entries=[]          
             self.name=name
-            self.writtenCells = {}
-            
-    def Add_Header(self, headerLines):
-        row = 0;
-        maxCol = 0;
-        for header in headerLines:
-            col = 0
-            for headerLine in header:
-                self.Write_String(row, col, headerLine)
-                col = col + 1
-                maxCol = max(col, maxCol)
-        self.currentCol = maxCol
-        
+            self.fontNames=[]
+            self.borderFormat = {}            
+            self.mergeable = set()
     
-    def Write_String(self, row, col, stringVal, formatValue):
-        # by putting chr(2) at the front of the string, google docs will not remove
-        # the leading spaces
-        tabString = " " * 4
-        stringVal = stringVal.replace("\t", tabString)
-        self.Inc_Max_Col(stringVal)
+    # fontNames lets us apply different fonts per column
+    # mergeable says whether we should merge all the rows to the 
+    # bottom of this group (we use it for names, all the rows
+    # associated with a name are merged)        
+    def Add_Header(self, headerLines, fontNames, mergeable):
+        self.entries.append(headerLines)
+        self.fontNames=fontNames
+        self.mergeable.update(mergeable)
         
-
-        stringVal = chr(2) + stringVal;
-        stringVal = stringVal.strip()
-        print(stringVal)
-        # storing it in col/row because we are going to walk down the columns & merge the cells
-        if (col not in self.writtenCells):
-            self.writtenCells[col] = []       
-        self.writtenCells[col].append((row, stringVal, formatValue))
-        self.worksheet.write_string(row, col, stringVal, formatValue)
-            
-    def Get_Indent_Format(self, indentSize):
-        if (indentSize not in self.indentFormat):
-            temp_format = self.workbook.add_format({'font_name':'Courier New'})
-            temp_format.set_font_size(11)
-            #temp_format.set_text_wrap()
-            #temp_format.set_shrink()
-            temp_format.set_align('top')
-            temp_format.set_indent(indentSize)
-            self.indentFormat[indentSize] = temp_format
-        return self.indentFormat[indentSize]
+    # Not the most generic name, but the needs to be called when we add the column zero block (student name for us).
```

Comment:
s/the needs/this needs?

Reply:
So, what is the flow for fixing these with git hub?  If I make the fix, do I then commit a new branch & create a new pull request?

Reply:
What I usually do is rebase the branch to make adjustments and then force push (`git push --force ...`) to update it. Unfortunately Github doesn't have a very sensible way of representing this and the previous diff (including this present comment thread) will become difficult to find. I have not found a better work flow for pre-merge review though.